db/migrate/20170501201633_add_canceled_to_order_detail.rb

```diff
@@ -4,10 +4,11 @@ def up
     add_column :order_details, :canceled_by, :integer
     add_column :order_details, :canceled_reason, :string
 
-    Reservation.where.not(order_detail_id: nil).find_each do |reservation|
-      reservation.order_detail.update_attributes(canceled_at: reservation.canceled_at,
+    Reservation.where.not(order_detail_id: nil, canceled_at: nil).find_each do |reservation|
+      reservation.order_detail.assign_attributes(canceled_at: reservation.canceled_at,
```

Comment:
I ended up putting the data change in the migration because it seemed important to make sure that the data change was definitely run before the old `reservation.canceled_*` columns were dropped. Putting it in a migration ensured that it would be impossible to miss migrating the data, where with a rake task, one could conceivably inadvertently skip running it before dropping the old columns and end up with data loss.

Coordinating running a rake task across four or five staging environments, three production environments, and everyone's dev environment seemed a little error-prone to me. Idk if it's worth changing from a migration at this point, but curious what your preferred alternative might be, Noel.